llvm/include/llvm/IR/RepoTicket.h

```diff
@@ -118,111 +118,150 @@ class TicketNode : public MDNode {
   friend class LLVMContextImpl;
   friend class MDNode;
 
-  struct CheckLinkageType {
-    using LT = std::underlying_type<GlobalValue::LinkageTypes>::type;
-    using ULT = std::make_unsigned<LT>::type;
+  template <unsigned Bitwidth, typename Type> struct CheckType {
+    using LT = typename std::underlying_type<Type>::type;
+    using ULT = typename std::make_unsigned<LT>::type;
     template <typename U> static constexpr U max() {
       return std::numeric_limits<U>::max();
     }
-    static constexpr bool isSafeCast() {
+    static bool isSafeCast(Type Value) {
       return (std::is_unsigned<LT>::value ? true : max<LT>() > 0) &&
-             max<ULT>() <= max<unsigned>();
+             max<ULT>() <= max<unsigned>() &&
+             static_cast<unsigned>(Value) <= (1u << Bitwidth);
+    }
+  };
+
+  struct Data32 {
+    static unsigned combine(GlobalValue::VisibilityTypes high,
+                            GlobalValue::LinkageTypes low) {
+      return static_cast<unsigned>(((uint64_t)high) << 16) | ((uint64_t)low);
+    }
+
+    static GlobalValue::VisibilityTypes high(unsigned combined) {
```

Comment:
`high()` doesn't tell the reader anything about the value, could we call it `visibility` or similar?

Reply:
Fixed by changing `high()` to `visibility()`.